common/djangoapps/track/middleware.py

```diff
@@ -204,3 +205,51 @@ def process_response(self, _request, response):
             pass
 
         return response
+
+
+    def truncate_openended_responses(self, event):
+        """
+        Truncates the tracking log entry for problems for which the
+        student response can be long winded.
+
+        event: dict, Constructed from POST and GET request params
+        """
+        event_json = json.loads(event)
+        event_post_keys = event_json['POST'].keys()
+        for key in event_post_keys:
+
+            # problem type with 'student_answer'
+            if key == 'student_answer' and isinstance(event_json['POST'].get('student_answer'), list):
+                truncated_answers = []
+                student_answer = event_json['POST'].get('student_answer')
+
+                # Loop round all the answers in the array and truncate each one, if necessary.
+                for answer in student_answer:
+                    truncated_answer = answer[:int(TRUNCATION_LENGTH/len(student_answer))]
+                    truncated_answers.append(truncated_answer)
+
+                # Put the truncated answers array back into the event dict.
+                event_json['POST']['student_answer'] = truncated_answers
+                return json.dumps(event_json)
+
```

Comment:
While not as bad as it is with class methods, the copious newlines here make it hard to visually group logical branches here. Could we drop them?